tests/test_validation_passive_context_budget.py

```
@@ -0,0 +1,438 @@
+    ) -> None:
+        """Traversal rejection must emit a diagnostic to stderr."""
+        target = {"path": "../escape", "max_tokens": 5000, "label": "Escape"}
+        measure_file(tmp_path, target)
+        captured = capsys.readouterr()
+        assert "traversal or absolute paths not allowed" in captured.err
+
+    def test_absolute_path_stderr_message(
+        self, tmp_path: Path, capsys: pytest.CaptureFixture[str],
+    ) -> None:
+        """Absolute path rejection must emit a diagnostic to stderr."""
+        target = {"path": "/tmp/nope", "max_tokens": 5000, "label": "Abs"}
```

Ruff (0.15.9)

[error] Probable insecure usage of temporary file or directory: "/tmp/nope"

(S108)